src/python/pants/backend/debian/rules_integration_test.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+from textwrap import dedent
+
+import pytest
+
+from pants.backend.debian.rules import package_debian_package
+from pants.backend.debian.target_types import DebianPackage
+from pants.backend.python import target_types_rules
+from pants.core.util_rules import source_files
+from pants.core.util_rules.source_files import SourceFilesRequest, SourceFiles
+from pants.engine.rules import QueryRule
+from pants.testutil.rule_runner import RuleRunner
+
+
+@pytest.fixture
+def rule_runner() -> RuleRunner:
+    return RuleRunner(
+        rules=[
+            *package_debian_package(),
+            *source_files.rules(),
+            *target_types_rules.rules(),
+            QueryRule(SourceFiles, (SourceFilesRequest,)),
```

Comment:
Here's an example of how to write tests for `package`: https://github.com/pantsbuild/pants/blob/main/src/python/pants/backend/go/goals/package_binary_integration_test.py

Some suggestions:

1. You'll want to use `rule_runner.request(BuiltPackage, [DebianPackageFieldSet(...)])`, like those Go tests.
2. To do that, you'll have to get the rule graph happy...which is by far the worst part of writing these tests. You will need to update this `QueryRule` appropriately, and also register the `rules()` for your production file as suggested above, _plus_ possibly any other rules used transitively by your code. Please feel free to ask for help
3. Then, you should be good to go! The `rule_runner.request()` will cause Pants to execute your `package_debian_package` rule using whatever input you give it

Reply:
Thanks, hugely helpful. Done!

Reply:
Nit: you can remove this `QueryRule` - it would only be necessary if you said `rule_runner.request(SourceFiles, [SourceFilesRequest()]`. It's not _wrong_ to have it, but it's unnecessary so a little distracting.